docs/specs/code/EXPORT_TRANSACTION_HISTORY_TO_FILE.md

```diff
@@ -0,0 +1,67 @@
+## Abstract
+
+Users want to export their transaction history into an Excel document.
+
+Default format gotta be XLSX, but optionally we might also support CSV, cuz it's more universal and might be imported into Excel as well as lots of other programs.
+
+Users should be able to chose the time period to export transaction for, like day from and day to. There also might be a list of default periods, e.g.: "this month", "last 30 days", "this quarter", "this year", "all". Default period selected might be "this month".
+
+## UI
+
+I suggest adding a new "Export to File" icon at the top-right of the transactions list. Something like this (general idea):
+
+![image](https://user-images.githubusercontent.com/5585355/50343034-8e94cc80-0536-11e9-9058-1af18be67f8a.png)
+
+When user clicks "Export to file" - new dialog window opens up:
+
+![image](https://user-images.githubusercontent.com/5585355/50346357-31534800-0543-11e9-949a-873bcae9b854.png)
+
+Dialog contains:
+1. Two date selectors: "from day", "to day"
+2. Dropdown-selector "Select period"
```

Comment:
There's this part in the spec:

> There also might be a list of default periods, e.g.: "this month", "last 30 days", "this quarter", "this year", "all". Default period selected might be "this month".

Idk, if we need "last 30 days", just added as an example )

I thought most popular periods would be "this month", "last month", "this quarter", "last quarter", but we can come up with any. Idk, if we should do months into the list of predefined periods. I mean, "this month" makes sense, because it might change from day to day, as well as "last month". That's what I mostly meant by "predefined periods", like when it's easier for Yoroi to quickly calculate those.

A specific month (e.g. "January 2018") might be a useful case, by it's a fixed value. I think it could be done like this: list "Period" contains **dynamic** values like:
1. "This month"
2. "Last month"
3. "This quarter"
4. "Last quarter"
5. "This year"
6. "Last year"
7. "All time"

But then also contains values:
8. "A month"
9. "Custom period"

When user selects any **dynamic** value - they can see date-selectors, but those are locked. And when they select "Custom period" - they get to set values manually. And when they select "A month" - they see only one selector that allows to choose a year and a month.

Reply:
But we can get along with only the "Custom period" for now, cuz if would allow to select any month, just by selecting first and last day. And then add "A month" later, if needed.